--- js/src/js.c~	2007-04-20 20:45:18.000000000 +0200
+++ js/src/js.c	2016-10-18 22:22:16.509762977 +0200
@@ -160,7 +160,7 @@
 #endif
     {
         char line[256];
-        fprintf(gOutFile, prompt);
+        fprintf(gOutFile, "%s", prompt);
         fflush(gOutFile);
         if (!fgets(line, sizeof line, file))
             return JS_FALSE;
--- js/src/config/Linux_All.mk~	2005-05-10 21:53:44.000000000 +0200
+++ js/src/config/Linux_All.mk	2019-10-17 22:09:03.383335550 +0200
@@ -43,7 +43,7 @@
 
 CC = gcc
 CCC = g++
-CFLAGS +=  -Wall -Wno-format
+CFLAGS +=  -Wall
 OS_CFLAGS = -DXP_UNIX -DSVR4 -DSYSV -D_BSD_SOURCE -DPOSIX_SOURCE -DHAVE_LOCALTIME_R
 
 RANLIB = echo
