--- js/src/Makefile.ref.orig	Tue Mar 19 05:28:16 2002
+++ js/src/Makefile.ref	Sat May 18 15:36:52 2002
@@ -56,7 +56,7 @@
 
 ifdef JS_THREADSAFE
 DEFINES += -DJS_THREADSAFE
-INCLUDES += -I../../dist/$(OBJDIR)/include
+INCLUDES += -I../../dist/$(OBJDIR)/include -I/usr/include/nspr
 ifdef USE_MSVC
 OTHER_LIBS += ../../dist/$(OBJDIR)/lib/libnspr${NSPR_LIBSUFFIX}.lib
 else
@@ -122,7 +122,7 @@
 ifdef JS_READLINE
 # For those platforms with the readline library installed.
 DEFINES += -DEDITLINE
-PROG_LIBS += -lreadline -ltermcap
+PROG_LIBS += -lreadline
 else
 ifdef JS_EDITLINE
 # Use the editline library, built locally.
@@ -281,6 +281,7 @@
 PROGRAM = $(OBJDIR)/js
 ifdef JS_PERLCONNECT
 PROG_LIBS += $(PERLLDFLAGS)
+OTHER_LIBS += $(PERLLDFLAGS)
 endif
 endif
 
@@ -291,7 +291,7 @@
 	link.exe -out:"$@" $(EXE_LINK_FLAGS) $^
 else
 $(PROGRAM): $(PROG_OBJS) $(LIBRARY) $(FDLIBM_LIBRARY)
-	$(CC) -o $@ $(CFLAGS) $(PROG_OBJS) $(LIBRARY) $(LDFLAGS) $(OTHER_LIBS) \
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $(PROG_OBJS) -L$(OBJDIR) -ljs $(OTHER_LIBS) \
 	    $(PROG_LIBS)
 endif
 
--- js/src/perlconnect/Makefile.PL.orig	Tue Aug 14 15:09:43 2001
+++ js/src/perlconnect/Makefile.PL	Sat May 18 16:32:49 2002
@@ -29,7 +29,7 @@
 include ../config.mk
 
 all:
-	@echo '$(OBJDIR)'
+	@echo '$(OBJDIR)' >&2
 eof
 
 if ($foo{'m'}) {
@@ -40,7 +40,7 @@
 	open FOO, ">tempmakefile";
 	print FOO $tmpmk;
 	close FOO;
-	my $objdir = `gmake -f tempmakefile`;
+	my $objdir = `make -f tempmakefile 2>&1 >/dev/null`;
 	unlink "tempmakefile";
 	$inc = "-I.. -I../$objdir";
 	$libpath = "-L../$objdir";
--- js/src/liveconnect/Makefile.ref.orig	Tue Mar 19 05:28:45 2002
+++ js/src/liveconnect/Makefile.ref	Sat May 18 18:40:34 2002
@@ -125,7 +125,7 @@
 
 ifdef JS_THREADSAFE
 DEFINES += -DJS_THREADSAFE
-INCLUDES += -I../../../dist/$(OBJDIR)/include
+INCLUDES += -I../../../dist/$(OBJDIR)/include -I/usr/include/nspr
 ifdef USE_MSVC
 OTHER_LIBS += $(DEPTH)/../../dist/$(OBJDIR)/lib/libnspr${NSPR_LIBSUFFIX}.lib
 else
@@ -159,5 +159,5 @@
 
 $(PROGRAM): $(OBJDIR)/$(PROG_FILE).o $(OBJDIR)/jsj_simpleapi.o
 	@$(MAKE_OBJDIR)
-	$(CC) -o $@ $(CFLAGS) $+ $(LIBRARY) ../$(OBJDIR)/libjs.a $(LDFLAGS) $(OTHER_LIBS)
+	$(CC) -o $@ $(CFLAGS) $(LDFLAGS) $+ -L$(OBJDIR) -ljsj $(OTHER_LIBS) $(PROG_LIBS) -Wl,-rpath=$(JDK)/jre/lib/i386 -Wl,-rpath=$(JDK)/jre/lib/i386/native_threads -Wl,-rpath=$(JDK)/jre/lib/i386/classic -Wl,-rpath=$(JDK)/jre/lib/i386/client
 endif
--- js/src/liveconnect/config/Linux_All.mk.orig	Tue Mar 19 05:28:53 2002
+++ js/src/liveconnect/config/Linux_All.mk	Sat May 18 18:51:49 2002
@@ -45,9 +45,10 @@
 # INCLUDES   += -I$(JDK)/include -I$(JDK)/include/solaris
 INCLUDES   += -I$(JDK)/include/linux -I$(JDK)/include
 
-OTHER_LIBS += -L$(JDK)/jre/lib/i386/native_threads
-OTHER_LIBS += -L$(JDK)/jre/lib/i386/classic
-OTHER_LIBS += -L$(JDK)/jre/lib/i386 -ljava -ljvm -lhpi
+PROG_LIBS += -L$(JDK)/jre/lib/i386/native_threads
+PROG_LIBS += -L$(JDK)/jre/lib/i386/classic
+PROG_LIBS += -L$(JDK)/jre/lib/i386/client
+PROG_LIBS += -L$(JDK)/jre/lib/i386 -ljava -ljvm -lhpi -lverify
 
 # To run lcshell with the above, built in a Mozilla tree with a local nspr:
 # LD_LIBRARY_PATH=../../../dist/lib:/share/builds/components/jdk/1.2.2/Linux/jre/lib/i386:/share/builds/components/jdk/1.2.2/Linux/jre/lib/i386/classic:/share/builds/components/jdk/1.2.2/Linux/jre/lib/i386/native_threads CLASSPATH=./classes/Linux_All_DBG.OBJ/js15lc30.jar lcshell
--- js/src/rules.mk.orig	2003-11-15 01:11:04.000000000 +0100
+++ js/src/rules.mk	2004-03-24 16:33:03.000000000 +0100
@@ -73,6 +73,9 @@
 	@$(MAKE_OBJDIR)
 	$(CC) -o $@ -c $(CFLAGS) $*.c
 
+$(LIB_OBJS): $(OBJDIR)/%.o: %.c
+	@$(MAKE_OBJDIR)
+	$(CC) -o $@ -fPIC -c $(CFLAGS) $*.c
 
 $(OBJDIR)/%.o: %.c
 	@$(MAKE_OBJDIR)
@@ -118,7 +118,7 @@
 	$(RANLIB) $@
 
 $(SHARED_LIBRARY): $(LIB_OBJS)
-	$(MKSHLIB) -o $@ $(LIB_OBJS) $(LDFLAGS) $(OTHER_LIBS)
+	$(MKSHLIB) $(LDFLAGS) -o $@ $(LIB_OBJS) $(OTHER_LIBS) -lm
 endif
 endif
 
